projects/RabbitMQ.Client.OpenTelemetry/TraceProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+using RabbitMQ.Client.OpenTelemetry;
+
+namespace OpenTelemetry.Trace
+{
+    public static class OpenTelemetryExtensions
+    {
+        internal static TextMapPropagator s_propagator = Propagators.DefaultTextMapPropagator;
+
+        public static TracerProviderBuilder AddRabbitMQ(this TracerProviderBuilder builder,
+            RabbitMQOpenTelemetryConfiguration configuration)
+        {
+            if (configuration.PropagateBaggage)
+            {
+                s_propagator = new CompositeTextMapPropagator(new TextMapPropagator[]
+                {
+                    new TraceContextPropagator(), new BaggagePropagator()
+                });
+            }
+            else
+            {
+                s_propagator = new TraceContextPropagator();
+            }
+
+            RabbitMQActivitySource.UseRoutingKeyAsOperationName = configuration.UseRoutingKeyAsOperationName;
+            RabbitMQActivitySource.ContextExtractor = OpenTelemetryContextExtractor;
+            RabbitMQActivitySource.ContextInjector = OpenTelemetryContextInjector;
+
+            if (configuration.IncludeSubscribers)
```

Comment:
I don't have any strong opinion here, but want to challenge the need for this flag (and `IncludePublishers`)
if users want more control, they can always call `AddSource` themselves.

I guess my broader question is when would users want to enable rabbitmq via an instrumentation library if all they really need to do is call
`builder.AddSource("RabbitMQ.Client.*")`

Reply:
> I don't have any strong opinion here, but want to challenge the need for this flag (and `IncludePublishers`) if users want more control, they can always call `AddSource` themselves.
> 
> I guess my broader question is when would users want to enable rabbitmq via an instrumentation library if all they really need to do is call `builder.AddSource("RabbitMQ.Client.*")`

The main reason for the library was to simplify the OTel configuration and setup for users (with the extension method) and to work around the current issue that since OTel and Activity Baggage propagation is different we do not have to make the `OpenTelemetry.Api` package a dependency on the RabbitMQ library itself.

Reply:
I'm honestly not sure whether the separation of allowing people to turn on/off subscriber or publisher. Is that actually a usecase?

I can understand adding it here if that's a usecase seeing as this is for simplification. My question is, should you allow people to just do one or the other?

Reply:
> I'm honestly not sure whether the separation of allowing people to turn on/off subscriber or publisher. Is that actually a usecase?
> 
> I can understand adding it here if that's a usecase seeing as this is for simplification. My question is, should you allow people to just do one or the other?

No idea, but I think it was you and @lmolkova that suggested I separated the ActivitySources in the earlier OTel pr: https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1261/files#r1392469382

I have no preference either way really

Reply:
Having different names is more future-proof, but also it costs nothing since you can enable sources with wildcard.

I didn't try to push you to remove the instrumentation library, was just trying to understand what'd be the better case for it.

The `Activity.Baggage` problem is an  interesting one. @martinjt do you know if there are any plans to make baggages work better together? I can create issues/ping people/etc - it'd be great if we could solve it for everyone without instrumentations trying to patch it in each case.

Reply:
> The `Activity.Baggage` problem is an interesting one. @martinjt do you know if there are any plans to make baggages work better together? I can create issues/ping people/etc - it'd be great if we could solve it for everyone without instrumentations trying to patch it in each case.

Here's an old issue that relates to it (CC @cijothomas): https://github.com/open-telemetry/opentelemetry-dotnet/issues/1842

Reply:
The Activity baggage vs OTel Baggage is still an unresolved problem! It'll eventually be fixed, though we don't have specifics or timeline. (its not likely coming in this year, there is nothing tracking it in .NET 9 release timeframe.)